doc/source/whatsnew/v1.2.0.rst

```diff
@@ -331,6 +331,7 @@ Plotting
 
 - Bug in :meth:`DataFrame.plot` was rotating xticklabels when ``subplots=True``, even if the x-axis wasn't an irregular time series (:issue:`29460`)
 - Bug in :meth:`DataFrame.plot` where a marker letter in the ``style`` keyword sometimes causes a ``ValueError`` (:issue:`21003`)
+- Twinned axes were losing their tick labels which should only happen to all but the last row or column of 'externally' shared axes (:issue:`33819`)
```

Comment:
Is "externally" shared axes a standard term?

Reply:
No, hence the quotes. MPL overloading the term "axes" is problem:

Twinned axes were losing their tick labels which should only happen to all but the last row or column of axes shared between axes (:issue:`33819`)

Reply:
I should add that this is explained more thoroughly in the linked Issue #33819